RecoEgamma/ElectronIdentification/plugins/ElectronRegressionValueMapProducer.cc

```diff
@@ -20,32 +20,190 @@
+                       const edm::EventSetup& iSetup,
+                       std::unordered_map<std::string,float>& float_vars,
+                       std::unordered_map<std::string,int>& int_vars ) {
+    LazyTools* tools = static_cast<LazyTools*>(tools_tocast);
+    
+    const auto& the_sc  = iEle->superCluster();
+    const auto& theseed = the_sc->seed();
+    
+    const int numberOfClusters =  the_sc->clusters().size();
+    const bool missing_clusters = !the_sc->clusters()[numberOfClusters-1].isAvailable();
+    
+    std::vector<float> vCov = tools->localCovariances( *theseed );
+    
+    const float eMax = tools->eMax( *theseed );
+    const float e2nd = tools->e2nd( *theseed );
+    const float eTop = tools->eTop( *theseed );
+    const float eLeft = tools->eLeft( *theseed );
+    const float eRight = tools->eRight( *theseed );
+    const float eBottom = tools->eBottom( *theseed );
+    
+    float dummy;
+    int iPhi;
+    int iEta;
+    float cryPhi;
+    float cryEta;
+    EcalClusterLocal _ecalLocal;
+    if (iEle->isEB()) 
+      _ecalLocal.localCoordsEB(*theseed, iSetup, cryEta, cryPhi, iEta, iPhi, dummy, dummy);
+    else 
+      _ecalLocal.localCoordsEE(*theseed, iSetup, cryEta, cryPhi, iEta, iPhi, dummy, dummy);
+    
+    double see = (isnan(vCov[0]) ? 0. : sqrt(vCov[0]));
+    double spp = (isnan(vCov[2]) ? 0. : sqrt(vCov[2]));
+    double sep;    
+    if (see*spp > 0)
+      sep = vCov[1] / (see * spp);
+    else if (vCov[1] > 0)
+      sep = 1.0;
+    else
+      sep = -1.0;
+    
+    set_map_val(k_sigmaIEtaIPhi,sep,float_vars);
+    set_map_val(k_eMax,eMax,float_vars);
+    set_map_val(k_e2nd,e2nd,float_vars);
+    set_map_val(k_eTop,eTop,float_vars);
+    set_map_val(k_eBottom,eBottom,float_vars);
+    set_map_val(k_eLeft,eLeft,float_vars);
+    set_map_val(k_eRight,eRight,float_vars);
+    set_map_val(k_cryPhi,cryPhi,float_vars);
+    set_map_val(k_cryEta,cryEta,float_vars);
+
+    set_map_val(k_iPhi,iPhi,int_vars);
+    set_map_val(k_iEta,iEta,int_vars);
+    
+    std::vector<float> _clusterRawEnergy;
+    _clusterRawEnergy.resize(std::max(3, numberOfClusters), 0);
+    std::vector<float> _clusterDEtaToSeed;
+    _clusterDEtaToSeed.resize(std::max(3, numberOfClusters), 0);
+    std::vector<float> _clusterDPhiToSeed;
+    _clusterDPhiToSeed.resize(std::max(3, numberOfClusters), 0);
+    float _clusterMaxDR     = 999.;
+    float _clusterMaxDRDPhi = 999.;
+    float _clusterMaxDRDEta = 999.;
+    float _clusterMaxDRRawEnergy = 0.;
+    
+    size_t iclus = 0;
+    float maxDR = 0;
+    edm::Ptr<reco::CaloCluster> pclus;
+    if( !missing_clusters ) {
+      // loop over all clusters that aren't the seed  
+      auto clusend = the_sc->clustersEnd();
+      for( auto clus = the_sc->clustersBegin(); clus != clusend; ++clus ) {
+        pclus = *clus;
+      
+        if(theseed == pclus ) 
+          continue;
+        _clusterRawEnergy[iclus]  = pclus->energy();
```

Comment:
note for posterity: this is the part of #11236, which actually changes the saved values; the rest is just refactoring